### DISCUSSION TITLE
Why does api.bsky.app have a fake DID?

### DISCUSSION THREAD

**Discussion by yamarten** [bluesky-social/atproto#2811] [Clarifications and "Why?"] [answered]
[In the PDS installer script](https://github.com/bluesky-social/pds/blob/main/installer.sh#L55), the DID of bsky appview is set to `did:web:api.bsky.app`. However, the DID document does not exist, and the following URL is 404.

https://api.bsky.app/.well-known/did.json

PDS seems to bypass DID resolution by embedding the DID and the URL in pairs.

https://github.com/bluesky-social/atproto/blob/85c85350d13fb4468c12045e086ad91733b4bcb3/packages/pds/src/api/app/bsky/notification/registerPush.ts#L29-L35

Why is it like this? Please let me know if it has a real DID.

As an aside, the reason for this question is that appview could not be explicitly specified with `atproto-proxy`. Even if a DID document exists, there is probably no service endpoint identifier for the bsky API in it, but I expect it can be replaced by [`bsky_notif`](https://github.com/bluesky-social/atproto/blob/85c85350d13fb4468c12045e086ad91733b4bcb3/packages/bsky/src/api/well-known.ts#L25).

**Comment by DavidBuchanan314** [comment]
+1, this issue is currently blocking correct implementation of `atproto-proxy` in either clients or PDSes

**Comment by yamarten** [comment]
As a supplement, after this question, bluesky-social/bsky-docs#198 indicated that the appview's service DID was TODO.

https://docs.bsky.app/docs/advanced-guides/api-directory#bluesky-services

Since it is unclear what is preventing this, this question is still open.

**Comment by bnewbold** [comment, marked as answer]
Yup! This is basically just tech debt we need to get around to.

**Comment by yamarten** [comment]
Currently, Bluesky appview provides its DID document.

https://api.bsky.app/.well-known/did.json
